catalyst_voices/lib/widgets/cards/role_chooser_card.dart

```diff
@@ -0,0 +1,241 @@
+const _grayscaleFilter = ColorFilter.matrix([
+  0.2126,
+  0.7152,
+  0.0722,
+  0,
+  0,
+  0.2126,
+  0.7152,
+  0.0722,
+  0,
+  0,
+  0.2126,
+  0.7152,
+  0.0722,
+  0,
+  0,
+  0,
+  0,
+  0,
+  1,
+  0,
+]);
+
+/// A role chooser card, responsible as a building block
+/// for any role selection, and available in both
+/// interactive and view-only mode.
+class RoleChooserCard extends StatelessWidget {
+  /// The current displaying value.
+  final bool value;
+
+  /// Needs to be a rasterized image.
+  final String imageUrl;
+
+  /// The text label displaying on the card.
+  final String label;
+
+  /// Locks the value and shows it as default, only the selected value appears.
+  final bool lockValueAsDefault;
+
+  /// Hides the "Learn More" link.
+  final bool isLearnMoreHidden;
+
+  /// Toggles view-only mode.
+  final bool isViewOnly;
+
+  /// A callback triggered when the role selection changes.
+  final ValueChanged<bool>? onChanged;
+
+  /// A callback triggered when the "Learn More" link is clicked.
+  final VoidCallback? onLearnMore;
+
+  const RoleChooserCard({
+    super.key,
+    required this.value,
+    required this.imageUrl,
+    required this.label,
+    this.lockValueAsDefault = false,
+    this.isLearnMoreHidden = false,
+    this.isViewOnly = false,
+    this.onChanged,
+    this.onLearnMore,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: 100,
+      decoration: isViewOnly
+          ? null
+          : BoxDecoration(
+              border: Border.all(
+                color: Theme.of(context).colors.outlineBorderVariant!,
+                width: 1,
+              ),
+              borderRadius: BorderRadius.circular(8),
+            ),
+      child: Padding(
+        padding: EdgeInsets.all(isViewOnly ? 8 : 12),
+        child: Row(
+          children: [
+            Column(
+              children: [
+                ColorFiltered(
+                  colorFilter: _grayscaleFilter,
+                  child: CatalystImage.asset(
+                    imageUrl,
+                    width: 70,
+                    height: 70,
+                    fit: BoxFit.cover,
+                  ),
+                ),
+              ],
+            ),
+            const SizedBox(width: 12),
+            Expanded(
+              child: Column(
+                children: [
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      Expanded(
+                        child: Text(
+                          overflow: TextOverflow.ellipsis,
+                          label,
+                          style: Theme.of(context).textTheme.titleSmall,
+                        ),
+                      ),
+                      if (!isLearnMoreHidden) ...[
+                        const SizedBox(width: 10),
+                        GestureDetector(
```

Comment:
this could be private widget at the. The `LinkText` will add `underline` so you would need to make parameter wheter to do it or not in `LinkText`

```dart
if (!isLearnMoreHidden) ...[
  const SizedBox(width: 10),
  _LearnMoreText(onTap: onLearnMore),
],

class _LearnMoreText extends StatelessWidget {
  final VoidCallback? onTap;

  const _LearnMoreText({
    this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return LinkText(
      context.l10n.learnMore,
      onTap: onTap,
      style: Theme.of(context).textTheme.labelMedium,
    );
  }
}
```